rust/noosphere-core/src/data/address.rs

```diff
@@ -99,15 +153,60 @@ impl LinkRecord {
                 }
             }
         }
-
         warn!("No facts contained a link!");
-
         None
     }
+
+    /// Return a base64-encoded string serialization of the
+    /// underlying [Ucan].
+    pub fn encode(&self) -> Result<String> {
+        self.0.encode()
+    }
+
+    /// Return a base64-encoded [Vec<u8>] of the underlying [Ucan].
+    pub fn as_bytes(&self) -> Result<Vec<u8>> {
+        Ok(Vec::try_from(self.encode()?)?)
+    }
 }
 
+impl ser::Serialize for LinkRecord {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        let encoded = self.encode().map_err(ser::Error::custom)?;
+        serializer.serialize_str(&encoded)
+    }
+}
+
+impl<'de> de::Deserialize<'de> for LinkRecord {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        let record = LinkRecord::try_from(s).map_err(de::Error::custom)?;
+        Ok(record)
+    }
+}
+
+/// [LinkRecord]s compare their [Jwt] representations
+/// for equality. If a record cannot be encoded as such,
+/// they will not be considered equal to any other record.
+impl PartialEq for LinkRecord {
```

Comment:
This is a pretty expensive equality check. Would you be willing to file an issue for us to implement `PartialEq` for [`Ucan` et al](
https://github.com/ucan-wg/rs-ucan/blob/main/ucan/src/ucan.rs#L20-L42C18) so that we can avoid the encoding cost?

Reply:
Agreed, will do!